crates/apub/src/fetcher/search.rs

```diff
@@ -18,28 +19,22 @@ use url::Url;
 /// which gets resolved to an URL.
 #[tracing::instrument(skip_all)]
 pub(crate) async fn search_query_to_object_id(
-  query: &str,
+  mut query: String,
```

Comment:
Instead of having this a a mutable param, I think it'd be better to just create a new var inside this function. immutable functions and all that.

Reply:
Unfortunately [String::remove](https://doc.rust-lang.org/std/string/struct.String.html#method.remove) only works on a mutable string, and I dont see a better alternative in this case. Sure I could declare a separate mut inside the function, but that wouldnt really change anything.

Reply:
It's still an immutable function.